core/storage/voiceover/gae_models.py

```diff
@@ -206,3 +217,152 @@ def get_export_policy(cls) -> Dict[str, base_models.EXPORT_POLICY]:
         return dict(super(cls, cls).get_export_policy(), **{
             'language_codes_mapping': base_models.EXPORT_POLICY.NOT_APPLICABLE
         })
+
+
+class VoiceArtistMetadataModel(base_models.BaseModel):
+    """The model stores manual voice artists' information with their
+    provided voiceovers metadata.
+    Instances of this class are keyed by the user ID.
+    """
+
+    # The user ID of the voiceover artist.
+    voice_artist_id = datastore_services.StringProperty(
+        indexed=True, required=True)
+    # Dictionary mapping language codes to nested dictionaries. Each
+    # nested dictionary contains the following key-value pairs:
+    #   -'language_accent_code' key holds the language accent code value.
+    #   -'exploration_id_to_content_ids' key maps exploration IDs to lists of
+    #       content IDs.
+    #   -'voiceovers' key contains a list of sample voiceovers associated
```

Comment:
Typeinfo is especially important here -- not clear whether these are voiceover urls or voiceover objects etc.

Explain if there are any limits here -- will there be at least 1 sample voiceover, or are empty lists possible? Any max limit?

Reply:
Updated, PTAL